lapce-data/src/config.rs

```diff
@@ -329,10 +329,18 @@ impl Themes {
     }
 }
 
+#[derive(FieldNames, Debug, Clone, Deserialize, Serialize, Default)]
+#[serde(rename_all = "kebab-case")]
+pub struct PluginsConfig {
+    #[field_names(desc = "Check for plugins automatically")]
+    pub auto_check: bool,
```

Comment:
What happens when this is false? The plugin list in the plugins panel can't be updated? I think it would be nice to add a manual update option in this case.

Reply:
> What happens when this is false? The plugin list in the plugins panel can't be updated? I think it would be nice to add a manual update option in this case.

Yes I agree. There should be a "Check Now" button if this setting is false.

In retrospection, this PR is a bit premature. I was only thinking of having the option to not do the fetch on starting up.

I wouldn't mind if this PR is sent back to the drawing board. I need to learn druid to add the "Check Now" button.

Reply:
You can always press the "Convert to draft" button ;)

Reply:
> You can always press the "Convert to draft" button ;)

Yeah I will think about it.

By the way, I noticed that we are not using the druid widgets such as button, checkbox, etc. For example, I am looking at https://github.com/lapce/lapce/blob/6e449b1d5cc62dbb2453602668d3b7dab24df91b/lapce-ui/src/keymap.rs#L451-L550

IIUC, this code draws a popup with a "Cancel" button and a a "Save" button. Then clicking the buttons is handled by https://github.com/lapce/lapce/blob/6e449b1d5cc62dbb2453602668d3b7dab24df91b/lapce-ui/src/keymap.rs#L58-L73

What're the reasons that we are creating the UI this way?

Reply:
@dzhou121 knows this